src/EditorFeatures/CSharpTest/IntroduceUsingStatement/IntroduceUsingStatementTests.cs

```diff
@@ -0,0 +1,519 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.IntroduceUsingStatement;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.IntroduceUsingStatement
+{
+    [Trait(Traits.Feature, Traits.Features.CodeActionsIntroduceUsingStatement)]
+    public sealed class IntroduceUsingStatementTests : AbstractCSharpCodeActionTest
+    {
+        protected override CodeRefactoringProvider CreateCodeRefactoringProvider(Workspace workspace, TestParameters parameters)
+            => new CSharpIntroduceUsingStatementCodeRefactoringProvider();
+
+        [Theory]
+        [InlineData("v[||]ar name = disposable;")]
+        [InlineData("var[||] name = disposable;")]
+        [InlineData("var [||]name = disposable;")]
+        [InlineData("var na[||]me = disposable;")]
+        [InlineData("var name[||] = disposable;")]
+        [InlineData("var name [||]= disposable;")]
+        [InlineData("var name =[||] disposable;")]
+        [InlineData("var name = [||]disposable;")]
+        [InlineData("[|var name = disposable;|]")]
+        [InlineData("var name = disposable[||];")]
+        [InlineData("var name = disposable;[||]")]
+        [InlineData("var name = disposable[||]")]
+        public async Task RefactoringIsAvailableForSelection(string declaration)
```

Comment:
@Neme12 i would be totally ok with you writing a 'selection' test in this manner.